lib/fizzy/utf8.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <string_view>
+
+namespace fizzy
+{
+bool utf8_validate(const uint8_t* start, const uint8_t* end) noexcept;
+
+inline bool utf8_validate(std::string_view input) noexcept
+{
+    return utf8_validate(reinterpret_cast<const uint8_t*>(input.begin()),
```

Comment:
This should work
```suggestion
    return utf8_validate(input.data(), input.data() + input.size());
```

Reply:
It doesn't, `string` uses signed chars.

Reply:
Still I don't think it is safe to do it like that, i.e. `reinterpreter_cast` a `string_view::const_iterator` to `const uint8_t*`. You need something like that:
```
const char* begin = input.begin();
utf8_validate(reinterpret_cast<const uint8_t*>(begin)
```

Reply:
Or make the parameter `bytes_view input` and construct `bytes_view` over vector's data in the caller.

I don't see the necessity for `reinterpret_cast` here.

Reply:
Actually this helper is not needed if #196 is merged first.